# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/src/xpcprivate.h
# Commit: a3fd24f5f0a7
# Full Hash: a3fd24f5f0a7bc094409fdbdbe1ef92fc459677b
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-11-23 21:49:00
# Regressor Bug: 1579367
# File Overlap Count: 1
# Description:
#   Bug 1579367 - Initialize XPCJSContext explicitly, after loading user prefs. r=kmag
#   
#   This way we get the correct values for start-up prefs in the parent process.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D51061
# ==============================================================================

diff -r e06f2a4de969 -r a3fd24f5f0a7 js/xpconnect/src/xpcprivate.h
--- a/js/xpconnect/src/xpcprivate.h	Fri Nov 22 09:48:58 2019 +0000
+++ b/js/xpconnect/src/xpcprivate.h	Fri Nov 22 12:40:17 2019 +0000
@@ -246,6 +246,8 @@
   // Called by module code on dll shutdown.
   static void ReleaseXPConnectSingleton();
 
+  static void InitJSContext();
+
   void RecordTraversal(void* p, nsISupports* s);
 
  protected:
@@ -258,7 +260,7 @@
   static nsXPConnect* gSelf;
   static bool gOnceAliveNowDead;
 
-  XPCJSRuntime* mRuntime;
+  XPCJSRuntime* mRuntime = nullptr;
   bool mShuttingDown;
 
   friend class nsIXPConnect;